ext-apiserver/pkg/controller/machine/controller.go

```diff
@@ -34,22 +39,94 @@ type MachineControllerImpl struct {
 
 	// lister indexes properties about Machine
 	lister listers.MachineLister
+
+	// lister indexes properties about Cluster
+	clusterLister listers.ClusterLister
+
+	actuator cloud.MachineActuator
+
+	clientSet *clientset.Clientset
 }
 
 // Init initializes the controller and is called by the generated code
 // Register watches for additional resource types here.
 func (c *MachineControllerImpl) Init(arguments sharedinformers.ControllerInitArguments) {
 	// Use the lister for indexing machines labels
 	c.lister = arguments.GetSharedInformers().Factory.Cluster().V1alpha1().Machines().Lister()
```

Comment:
Since our objects are going to be namespace scoped https://github.com/kubernetes/kube-deploy/issues/463 , do we need to do anything to make sure this listers will list objects in the namespace we intend? (I am guessing said namespace is default for now to match line 66

Reply:
Lister().List() will have a chance to specify the selector
Lister().Machines() have a parameter to select namespace. The caller of those method will need to specify appropriate namespace. I don't have a solid idea who/where to filter around namespace yet.

Reply:
I am going to make an educated guess that when we fetch the cluster in getCluster(), we should be fetching from the namespace we are managing. You can also punt to figuring out this namespace stuff in a different review if you would prefer to open an issue to track that work. :)

Reply:
Yes, the current implementation are always fetching from default namespace. I will open an issue to track how we associate namespace with controller.

Reply:
https://github.com/kubernetes/kube-deploy/issues/520